clippy_lints/src/hash_borrow_str_semantics.rs

```diff
@@ -0,0 +1,103 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::ty::implements_trait;
+use rustc_hir::def::{DefKind, Res};
+use rustc_hir::{Item, ItemKind, Path, TraitRef};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::Ty;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::symbol::sym;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// This lint is concerned with the semantics of Borrow and Hash for a
+    /// type that implements all three of Hash, Borrow<str> and Borrow<[u8]>
```

Comment:
Can you put the trait references between ticks, like `Hash`? There's a lint for this, but it doesn't seem smart enough to trigger here